src/main/java/com/seed/api/core/UserService.java

```diff
@@ -1,29 +1,136 @@
 package com.seed.api.core;
 
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.handler.CorsHandler;
+import io.vertx.core.http.HttpMethod;
+
+import java.util.Set;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import io.vertx.ext.web.Router;
-
-/**
- * The user service class.
- */
 public class UserService extends ServicesService {
 
     /**
      * The logger utils class.
      */
     private Logger logger = LoggerFactory.getLogger(
-      UserService.class.getName());
+    UserService.class.getName()
+    );
+
 
     /**
-     * Sets routes for the http server.
+     * Sets routes for the http server related
+     * to user management (login and registration).
      * @param router The router used to set paths.
      */
+
     protected void setUserRoutes(final Router router) {
+        // Enable parsing of request body for POST requests
+        router.route().handler(BodyHandler.create());
+
+        // CORS configuration to allow requests from any origin
+        // Create the CORS handler with "*"
+        // as allowed origin and POST as allowed method
+
+
+CorsHandler corsHandler = CorsHandler.create("*")
+                                     .allowedMethods(Set.of(HttpMethod.POST));
+
+
+// Attach the CORS handler to the router
+router.route().handler(corsHandler);
+
+
+        // Route for user registration
+        router.post("/register").handler(this::handleUserRegistration);
+
+        // Route for user login
+        router.post("/login").handler(this::handleUserLogin);
 
-      // Set the routes for location.
-      this.setServicesRoutes(router);
+        // Set the routes for location
+        //(if there are any additional routes for user management)
+        this.setServicesRoutes(router);
     }
 
+    /**
+     * Handler for user registration.
+     * @param routingContext The routing context.
+     */
+    private void handleUserRegistration(final RoutingContext routingContext) {
+        // Extract user registration data from the request body
+        JsonObject registrationData = routingContext.getBodyAsJson();
+
+        // Extract individual fields from the registration data
+        String fullName = registrationData.getString("full_name");
+        String phoneNumber = registrationData.getString("phone_number");
+        String email = registrationData.getString("email");
+        String password = registrationData.getString("password");
+        String confirmPassword = registrationData.getString("confirm_password");

```

Comment:
You don't need confirm password. Its the front end that needs it for validation.